### PR TITLE
Dynamically add java handlers - UI changes

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js [hunk 1]
```diff
@@ -17,6 +17,7 @@
  */
 
 var javaPolicyArray = new Array(); //contains Java Polices list
+var javaPolicyIndexArray = []; //used to maintain the selected Java Policy ID's list
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js]: Comments should start with a capital letter.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js [hunk 2]
```diff
@@ -56,43 +57,58 @@ function loadAvailableJavaPolicies(applicationUUID, isGlobalPolicy) {
...
+
     //draw java policy items
     $("#javaPolicy_tbody").trigger("draw");
 
 
-
 }
+
+
+//this event triggers when an dynamic optional java policy get clicked. Need to maintain the selected policy id list
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js]: All the comments should start with a capital letter.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/resource-add.js [hunk 3]
```diff
@@ -61,8 +61,8 @@ $( document ).ready(function() {
         }
     });
 
-    //load all available java policy list from DB
-    loadAvailableJavaPolicies(null,true);
+    //load all available global (applicaiton level) java policy list from DB
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/resource-add.js]: Comments should start with a capital letter.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/resource-edit.js [hunk 4]
```diff
@@ -73,6 +73,8 @@ $( document ).ready(function() {
         }
     });
 
+    //load global (application level) dynamic optional java policies
```
Comment (rushmin) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/resource-edit.js]: Comments should start with a capital letter.

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js [hunk 5]
```diff
@@ -56,43 +57,58 @@ function loadAvailableJavaPolicies(applicationUUID, isGlobalPolicy) {
  * draw java policies list
  */
 function updateJavaPolicyPartial() {
-    var enabledStatus;  //if policy is mandatory make it wont allow user to un-tick the particular java policy
-    var checkedStatus; //if policy is mandatory always tick the java policy
-
+    var checkedStatus = "";
```
Comment (dinusha-dilrukshi) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js]: No need to initialize to "" right ?

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js (hunk 1 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js]: commit# https://github.com/wso2/carbon-appmgt/commit/40182a44e29db392d430df70a9a168a5926320d2

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js (hunk 2 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js]: commit# https://github.com/wso2/carbon-appmgt/commit/40182a44e29db392d430df70a9a168a5926320d2

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/resource-add.js (hunk 3 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/resource-add.js]: commit# https://github.com/wso2/carbon-appmgt/commit/40182a44e29db392d430df70a9a168a5926320d2

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/resource-edit.js (hunk 4 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/resource-edit.js]: commit# https://github.com/wso2/carbon-appmgt/commit/40182a44e29db392d430df70a9a168a5926320d2

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js (hunk 5 above)
Comment (lahiru823) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/java.policy.js]: commit# https://github.com/wso2/carbon-appmgt/commit/40182a44e29db392d430df70a9a168a5926320d2

**Pull request merged**